### DISCUSSION TITLE
Help with Header-Based API Versioning

### DISCUSSION THREAD

**Discussion by MBoui** [salvo-rs/salvo#725] [Q&A]
Hi,

I'm implementing header-based API versioning using the `x-api-version` header.
My goals are:
- **Routing**: Redirect requests to the appropriate handler (`handler-1` for `version-1`, `handler-2` for `version-2`) based on the header value.
- **Clean URL**: Maintain the same endpoint `/my-endpoint` for all versions.
- **Defaulting**: Route requests without the header (or invalid versions) to the latest handler (`handler-2`).

I'd appreciate assistance in determining the best way to achieve this.

Thanks guys!

**Comment by chrislearn** [comment]
like this?

```rust
 Router::with_path("my-endpoin").push(
        Router::with_filter_fn(|req, _|{
            req.header::<String>("x-api-version").as_deref() == Some("version1")
        }).get(version1)
    ).push(
        // This is for header has version 2 and without header.
        Router::new().get(version2)
    )
```

If you like salvo, give it a star, thank you!
